apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,107 @@
+import Post from '../models/post';
+
+
+const getPosts = async (req, res) => {
+  try {
+    const posts = await Post.find();
+    res.status(200).json(posts);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+const getPostsByCategory = async (req, res) => {
+  const { category } = req.params;
+  try {
+    const post = await Post.findById(category);
```

Comment:
This will try to find by post ID instead of category ID, to search for category you can do the following
```
Post.find({ category })
```
This will tell MongoDB to find all the records where category equals the category ID you are passing